planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/util.cpp

```diff
@@ -50,6 +51,81 @@ using autoware_utils::create_marker_color;
 using autoware_utils::create_marker_scale;
 using autoware_utils::create_point;
 
+SegmentRtree extract_uncrossable_segments(
+  const lanelet::LaneletMap & lanelet_map, const Point & ego_point, const double max_distance)
+{
+  SegmentRtree uncrossable_segments_in_range;
+  LineString2d line;
+  const auto ego_p = Point2d{ego_point.x, ego_point.y};
+  for (const auto & ls : lanelet_map.lineStringLayer) {
+    if (has_types(ls, {"road_border"})) {
+      line.clear();
+      for (const auto & p : ls) line.push_back(Point2d{p.x(), p.y()});
+      for (auto segment_idx = 0LU; segment_idx + 1 < line.size(); ++segment_idx) {
+        Segment2d segment = {line[segment_idx], line[segment_idx + 1]};
+        if (boost::geometry::distance(segment, ego_p) < max_distance) {
+          uncrossable_segments_in_range.insert(segment);
+        }
+      }
+    }
+  }
+  return uncrossable_segments_in_range;
+}
```

Comment:
```suggestion
SegmentRtree extract_uncrossable_segments(
  const lanelet::LaneletMap & lanelet_map, const Polygon2d & extraction_polygon)
{
  SegmentRtree uncrossable_segments_in_range;

  LineString2d line;
  for (const auto & ls : lanelet_map.lineStringLayer) {
    if (has_types(ls, {"road_border"})) {
      line.clear();
      for (const auto & p : ls) line.push_back(Point2d{p.x(), p.y()});
      for (auto segment_idx = 0LU; segment_idx + 1 < line.size(); ++segment_idx) {
        Segment2d segment = {line[segment_idx], line[segment_idx + 1]};
        if (boost::geometry::intersects(segment, extraction_polygon)) {
          uncrossable_segments_in_range.insert(segment);
        }
      }
    }
  }
  return uncrossable_segments_in_range;
}
```

If using extract_polygon instead of range